gcp_variant_transforms/transforms/merge_headers.py

```diff
@@ -13,26 +13,174 @@
 
-__all__ = ['MergeHeaders']
+from gcp_variant_transforms.beam_io.vcf_header_io import VcfHeader
+
+__all__ = ['HeaderConflictResolver', 'HeaderMerger', 'MergeHeaders']
+
+class HeaderConflictResolver(object):
+  """A class for resolving mistmatch between header fields (e.g. type, num).
+  """
+
+  def __init__(self, split_alternate_allele_info_fields=True):
+    """Initialize the class.
+
+    Args:
+     split_alternate_allele_info_fields (bool): if true, all INFO fields with
+        `Number=A` (i.e. one value for each alternate allele) will be stored
+        underthe `alternate_bases` record. If false, they will be stored with
+        the rest of the INFO fields as a repeated fields, hence compatible with
+        `Number=.` for example.
+     """
+    self._split_alternate_allele_info_fields = (
+        split_alternate_allele_info_fields)
+
+  def resolve(self, vcf_field_key, first_vcf_field_value,
+              second_vcf_field_value):
+    """Returns resolution if the conflict between the given field values is
+    resolvable.
+
+    Raises:
+      ValueError: if there is nothing to resolve (values are equal), or
+        the conflict cannot be resolved.
+    Args:
+      vcf_field_key (string): field key in VCF header.
+      first_vcf_field_value (integer or string): first field value.
+      second_vcf_field_value (integer or string): second field value.
+   """
+    if vcf_field_key == 'type':
+      return self._resolve_type(first_vcf_field_value, second_vcf_field_value)
+    elif vcf_field_key == 'num':
+      return self._resolve_number(first_vcf_field_value, second_vcf_field_value)
+    else:
+      # We only care about conflicts in 'num' and 'type' fields.
+      return first_vcf_field_value
+
+  def _resolve_type(self, first, second):
+    if first == second:
+      raise ValueError('Types are the same. Nothing to resolve: {}, {}'
+                       .format(first, second))
+    elif (first in ('Integer', 'Float') and second in ('Integer', 'Float')):
+      return 'Float'
+    else:
+      raise ValueError('Incompatible values cannot be resolved: '
+                       '{}, {}'.format(first, second))
+
+  def _resolve_number(self, first, second):
+    if first == second:
+      raise ValueError('Numbers are the same. Nothing to resolve: {}, {}'
+                       .format(first, second))
+    elif (self._is_bigquery_field_repeated(first) and
+          self._is_bigquery_field_repeated(second)):
+      # Unknown number of values
+      return None
+    else:
+      raise ValueError('Incompatible numbers cannot be resolved: '
+                       '{}, {}'.format(first, second))
+
+  def _is_bigquery_field_repeated(self, vcf_num):
+    """Returns true if the corresponding field in bigquery schema is repeated.
+
+    Args:
+      vcf_num (integer): value of field `Number` in VCF header.
+    """
+    if vcf_num in (0, 1):
+      return False
+    elif vcf_num == -1 and self._split_alternate_allele_info_fields:
```

Comment:
Please use vcf.parser.field_counts['A'] instead of -1

Reply:
Done